vortex-file/src/write/stats_accumulator.rs

```diff
@@ -64,20 +42,22 @@ impl StatsAccumulator {
         Ok(())
     }
 
-    pub fn into_array(mut self) -> VortexResult<Option<ArrayData>> {
-        let mut names = vec![];
-        let mut fields = vec![];
+    pub fn into_array(mut self) -> VortexResult<Option<(ArrayData, Vec<Stat>)>> {
```

Comment:
I don't like passing tuples around like this. Should we have a `StatsTable` struct that wraps the array and the vec?